be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -334,6 +336,10 @@ class SegmentIterator final : public ChunkIterator {
 
     std::unordered_map<ColumnId, ColumnAccessPath*> _column_access_paths;
     std::unordered_map<ColumnId, ColumnAccessPath*> _predicate_column_access_paths;
+
+    // All columns share the same stream in shared data,
+    // when enable_lake_io_coalesce is true and segment_size is less than lake_small_segment_file_threshold_size
+    std::unique_ptr<io::SharedBufferedInputStream> _shared_buffer_input_stream = nullptr;
```

Comment:
Remove this field and use `std::shared_ptr` for `_column_files`, if the file is small, just make `_column_files` share the same instance of SeekableInputStream.

Reply:
`_column_files` share the same instance of SeekableInputStream is ok, but the field  `_shared_buffer_input_stream` is still needed, because  `_column_files` may contain `delta_column_reader` so that we can not find the instance of SeekableInputStream from `_column_files` in O(1)  when we `init_column`.